pointercrate-user-api/src/endpoints/auth.rs

```diff
@@ -2,40 +2,89 @@ use crate::{
     auth::{BasicAuth, TokenAuth},
     ratelimits::UserRatelimits,
 };
-use pointercrate_core::{etag::Taggable, pool::PointercratePool};
+use pointercrate_core::{error::CoreError, etag::Taggable, pool::PointercratePool};
 use pointercrate_core_api::{
     error::Result,
     etag::{Precondition, Tagged},
     response::Response2,
 };
-use pointercrate_user::{error::UserError, AuthenticatedUser, PatchMe, Registration, User};
+use pointercrate_user::{error::UserError, AuthenticatedUser, PatchMe, User};
 use rocket::{
-    http::Status,
+    http::{Cookie, CookieJar, SameSite, Status},
     serde::json::{serde_json, Json},
     State,
 };
 use std::net::IpAddr;
 
-#[rocket::post("/register", data = "<body>")]
-pub async fn register(
-    ip: IpAddr, body: Json<Registration>, ratelimits: &State<UserRatelimits>, pool: &State<PointercratePool>,
-) -> Result<Response2<Tagged<User>>> {
+#[rocket::get("/authorize?<legacy>")]
+pub async fn authorize(
+    ip: IpAddr, ratelimits: &State<UserRatelimits>, legacy: Option<&str>, cookies: &CookieJar<'_>,
+) -> Result<Response2<()>> {
+    ratelimits.login_attempts(ip)?;
+
+    if legacy.is_some() {
+        let legacy_cookie = Cookie::build(("legacy", "true"))
+            .http_only(true)
+            .same_site(SameSite::Strict)
+            .path("/");
+
+        cookies.add(legacy_cookie);
+    }
```

Comment:
Instead of using a cookie, can we somehow give this information to google, and have google return it to us when it calls our callback? Maybe via query parameter?